### PR TITLE
tests: update reference tests

### PR DESCRIPTION
**Pull request by holiman**
Updates the tests to the most recent tip.

### CODE REVIEW

File: tests/state_test.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func TestState(t *testing.T) {
 				t.Run(key+"/snap", func(t *testing.T) {
 					withTrace(t, test.gasLimit(subtest), func(vmconfig vm.Config) error {
 						snaps, statedb, err := test.Run(subtest, vmconfig, true)
-						if snaps != nil && statedb != nil {
+						if err == nil && snaps != nil && statedb != nil {
```
Comment (karalabe) [inline tests/state_test.go:91]: Butt why?

Comment (holiman) [reply tests/state_test.go:91]: If all transactions are rejected, the test.Run returns an error, and no snaps. And the `snaps.Journal` will then return a different error, obscuring the first one

Comment (karalabe) [reply tests/state_test.go:91]: Should we perhaps delete `snaps != nil && statedb != nil` then? I just don't want to end up with skipping the snap clause altogether.

Comment (holiman) [reply tests/state_test.go:91]: Yes, that should be fine. Let's test it

**Review by karalabe — APPROVED** [review]
LGTM

**Pull request merged**